controllers/fleetnotification/fleetnotification_controller.go

```diff
@@ -55,21 +58,38 @@ func (r *ManagedFleetNotificationReconciler) Reconcile(ctx context.Context, requ
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling FleetNotification")
 
-	// TODO(user): your logic here
 	nr := ocmagentv1alpha1.ManagedFleetNotificationRecord{}
 
 	err := r.Client.Get(ctx, request.NamespacedName, &nr)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	for n, rn := range nr.Status.NotificationRecordByName {
+		resendWait := rn.ResendWait
+		for i, ri := range rn.NotificationRecordItems {
+			// Consider the record is stale if the lastSendTime is older than resendWait + 15 days
+			eol := ri.LastTransitionTime.Time.Add(time.Duration(resendWait+NotificationRecordStaleTimeoutInHour) * time.Hour)
+			if time.Now().After(eol) {
+				log.Info(fmt.Sprintf("NotificationRecord for notification %s and hostedcluster %s has not been updated "+
+					"for a while and considered as stale, cleaning up...", rn.NotificationName, ri.HostedClusterID))
```

Comment:
`s/for a while/for $(resendWait+NotificationRecordStaleTimeoutInHour) * time.Hour)/` seems better and more specific?